src/Microsoft.IdentityModel.Xml/KeyInfo.cs

```diff
@@ -66,6 +67,10 @@ public KeyInfo(SecurityKey key)
                 CertificateData = Convert.ToBase64String(x509Key.Certificate.RawData);
                 Kid = x509Key.Certificate.Thumbprint;
             }
+            else if (key is RsaSecurityKey rsaKey)
```

Comment:
There is a possibility that rsaKey.Parameters is null, when the constructor that takes an RSA as a parameter is used.
In this case you will have to use rsaKey.RSA.ExportParameters(false) to obtain the parameters.

Reply:
Done!

RSAParameters is a struct, so it can't actually be null. However, I think that checking whether it equals the default value accomplishes the same purpose.